Yoast/javascript
# Create Social Preview Editor component

Also check Yoast/wordpress-seo-premium#2736

## Summary


This PR can be summarized in the following changelog entry:

* [social-metadata-previews] Adds a SocialPreviewEditor component, which combines a SocialPreview with the fields to edit it.

## Relevant technical choices:

* The Caret styles are implemented in different components in different ways. @Xyfi and I have tried extracting the logic in this PR, but the rabbit-hole ran too deep. We will refactor once there is time for technical debt issues.
* Image updating is fixed in a different PR.
* Carets have been removed from the Social Previews, are now only on forms.
* Hover and "click to focus" functionality remains in both previews and forms.
* Already placed in Premium in https://github.com/Yoast/wordpress-seo-premium/pull/2736 

## Test instructions

This PR can be tested by following these steps:

* Build the monorepo components app, and verify that the components on the `Social Preview` tab work as intended. (hovering, carets, etc).
* Checkout premium branch `533-social-preview-editor` https://github.com/Yoast/wordpress-seo-premium/pull/2736, where we've placed the editor in its fill.
* `yarn link-monorepo` in the root.
* Because `link-monorepo` does not work in the premium folder, you have to do this the old fashioned way:
	* in the `wordpress-seo-premium/premium` folder, run the following:
		`yarn link "@yoast/social-metadata-previews"`
		`yarn link "@yoast/social-metadata-forms"`
* Now, build everything. In the root you can build or run de dev-server, in the premium folder you'll have to `grunt build`.
* When all is done, check the premium plugin in your browser. Edit a post, go to the social react tab, verify that it works as expected (except for the image setting, fixed in different PR).

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes #533
